### PR TITLE
Feature/cleanup and operators

### PR DESCRIPTION
**Pull request by FelixKahle**
This pull request refactors the handling of feasible space windows for dock allocation requests, expanding support from a single interval to multiple intervals throughout the codebase. This change increases modeling flexibility and improves correctness for both request construction and validation. Additionally, the PR updates related error types, test cases, and documentation to support the new multi-window logic. It also introduces minor dependency and configuration updates.

### Core modeling changes

* The `Request` type now stores and operates on a vector of `SpaceInterval` objects (`feasible_space_windows`) instead of a single `SpaceInterval` (`feasible_space_window`). This affects construction, accessors, and display logic in `crates/dock-alloc-model/src/req.rs`, with corresponding changes to `AnyRequest`, `AnyRequestRef`, and their methods. [[1]](diffhunk://#diff-6fc1277c19e81c5dfbc5eb84240d041ff5d9eda314003cab28ff721b66e573e0L73-R73) [[2]](diffhunk://#diff-6fc1277c19e81c5dfbc5eb84240d041ff5d9eda314003cab28ff721b66e573e0L88-R93) [[3]](diffhunk://#diff-6fc1277c19e81c5dfbc5eb84240d041ff5d9eda314003cab28ff721b66e573e0L153-R151) [[4]](diffhunk://#diff-6fc1277c19e81c5dfbc5eb84240d041ff5d9eda314003cab28ff721b66e573e0R179-R190) [[5]](diffhunk://#diff-6fc1277c19e81c5dfbc5eb84240d041ff5d9eda314003cab28ff721b66e573e0L195-R199) [[6]](diffhunk://#diff-6fc1277c19e81c5dfbc5eb84240d041ff5d9eda314003cab28ff721b66e573e0L267-R274) [[7]](diffhunk://#diff-6fc1277c19e81c5dfbc5eb84240d041ff5d9eda314003cab28ff721b66e573e0L368-R375)
* All validation and error handling for space windows (in request construction, problem building, and solution validation) now iterate over all provided windows, ensuring each is long enough and checking assignment containment against all windows. [[1]](diffhunk://#diff-5ae57639faca574431b1f1ff99958b5e02a3934654e3b1b6a159ad1cc91ba4eaL292-R299) [[2]](diffhunk://#diff-7c7fcbcecf2478e8cc297d57fbf689dc0b2d81110e68789cf7fdf676759268b4L303-R306) [[3]](diffhunk://#diff-6fc1277c19e81c5dfbc5eb84240d041ff5d9eda314003cab28ff721b66e573e0L88-R93)

### Error and display updates

* The `AssignmentOutsideSpaceWindowError` type now stores and displays a vector of space windows, updating its constructor, accessors, and `Display` implementation to show all relevant intervals. [[1]](diffhunk://#diff-cdf58d95ff31a9e43f8fc524786246de473c685b494412bfaa74fbe26482a7a0L123-R139) [[2]](diffhunk://#diff-cdf58d95ff31a9e43f8fc524786246de473c685b494412bfaa74fbe26482a7a0L150-R151) [[3]](diffhunk://#diff-cdf58d95ff31a9e43f8fc524786246de473c685b494412bfaa74fbe26482a7a0R161-R170)
* The `ProblemBuildError` enum removes unnecessary trait bounds and updates to support the new error type.

### Test and generator updates

* All tests and generators are updated to use vectors of space windows, with new test cases verifying correct behavior for multiple windows, edge cases, and error handling. [[1]](diffhunk://#diff-59d9f0911aa435f4cc8f7b4559be54fe0bff156cc013ad8f6af088fc7ff49cb1L659-R662) [[2]](diffhunk://#diff-d33e65a1bedfdafe13966e376d1693cb429729e7002b2aa12ad114e1bf075c91L1292-R1292) [[3]](diffhunk://#diff-d33e65a1bedfdafe13966e376d1693cb429729e7002b2aa12ad114e1bf075c91L1332-R1332) [[4]](diffhunk://#diff-d33e65a1bedfdafe13966e376d1693cb429729e7002b2aa12ad114e1bf075c91L1408-R1413) [[5]](diffhunk://#diff-d33e65a1bedfdafe13966e376d1693cb429729e7002b2aa12ad114e1bf075c91L1447-R1457) [[6]](diffhunk://#diff-5ae57639faca574431b1f1ff99958b5e02a3934654e3b1b6a159ad1cc91ba4eaL490-R497) [[7]](diffhunk://#diff-5ae57639faca574431b1f1ff99958b5e02a3934654e3b1b6a159ad1cc91ba4eaL511-R521) [[8]](diffhunk://#diff-6fc1277c19e81c5dfbc5eb84240d041ff5d9eda314003cab28ff721b66e573e0L417-R550)

### Configuration and dependency updates

* The dock allocation core crate adds the `rand` dependency and updates the `SolverVariable` trait to require `SampleUniform`, improving random sampling support. [[1]](diffhunk://#diff-4f4005f83a5521980150cd29ef148beeea9f2b0226b2043d042e5a7f2ad3127fR29) [[2]](diffhunk://#diff-7f9a3c51b90a47ba55df3882bb780662f2364cdf7ffd05dd639cdb69c455ba00R23) [[3]](diffhunk://#diff-7f9a3c51b90a47ba55df3882bb780662f2364cdf7ffd05dd639cdb69c455ba00L37-R46)
* Problem generator parameters are adjusted for more realistic test scenarios (e.g., reducing horizon and request counts).

### CODE REVIEW

File: crates/dock-alloc-solver/src/framework/iter.rs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                self.idx += 1;
+
+                if let Some(swin) = self.space_search.intersection(&wf)
+                    && self.twin.duration() >= self.p && swin.measure() >= self.len {
+                        self.cur = Some(
+                            self.berth
+                                .iter_free_regions(self.twin, self.p, self.len, swin),
+                        );
+                        break;
+                    }
```
[the remark this answers is not in the document]
Comment (FelixKahle) [reply crates/dock-alloc-solver/src/framework/iter.rs:188]: We keept it explicit per operator for now

File: crates/dock-alloc-solver/src/framework/iter.rs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                }
+                self.cur = None;
+            }
+
+            while self.idx < self.windows.len() {
+                let wf = self.windows[self.idx];
+                self.idx += 1;
+
+                if let Some(swin) = self.space_search.intersection(&wf)
+                    && self.twin.duration() >= self.p && swin.measure() >= self.len {
```
[the remark this answers is not in the document]
Comment (FelixKahle) [reply crates/dock-alloc-solver/src/framework/iter.rs:147]: Happens. We keep it
